traffic_editor/include/traffic_editor/building_level.h

```diff
@@ -63,6 +64,8 @@ class BuildingLevel : public Level
 
   QPixmap floorplan_pixmap;
 
+  mutable crowd_sim::CrowdSimImplPtr crowd_sim_impl;
```

Comment:
I find mutable to be a bit of a yellow flag at times, is there cases in which the crowd sim behavior is changed inside a `const` function?

Reply:
There are. Some properties in the object being pointed to get changed. I just keep it that way (Fred's version).

Reply:
Since it's a shared pointer changing anything in the object pointed at shouldn't cause a compile time error, it should only happen if the value of the pointer itself changes.
Maybe the functions that change the pointer itself shouldn't be `const`? Do you have any example of functions that give compile errors when `mutable` is removed?